lms/djangoapps/courseware/field_overrides.py

```diff
@@ -171,83 +255,84 @@ def default(self, block, name):
+        super(OverrideModulestoreFieldData, self).__init__(None, fallback, providers)
 
-    A `FieldOverrideProvider` implementation is only responsible for looking up
-    field overrides. To set overrides, there will be a domain specific API for
-    the concrete override implementation being used.
+
+class FieldOverrideModulestoreWrapper(object):
+    """Apply field overrides to XBlocks.
+
+    Extends read-only functionality.
     """
-    __metaclass__ = ABCMeta
 
-    def __init__(self, user):
-        self.user = user
+    def __init__(self, modulestore):
+        """Wrap the provided modulestore"""
+        self.__dict__['_modulestore'] = modulestore
 
-    @abstractmethod
-    def get(self, block, name, default):  # pragma no cover
-        """
-        Look for an override value for the field named `name` in `block`.
-        Returns the overridden value or `default` if no override is found.
-        """
-        raise NotImplementedError
+    def __getattr__(self, name):
+        """Look up missing attributes on the wrapped modulestore"""
+        return getattr(self._modulestore, name)
 
-    @abstractmethod
-    def enabled_for(self, course):  # pragma no cover
-        """
-        Return True if this provider should be enabled for a given course,
-        and False otherwise.
+    def __setattr__(self, name, value):
+        """Set attributes on the wrapped modulestore"""
+        setattr(self._modulestore, name, value)
 
-        Concrete implementations are responsible for implementing this method.
+    def __delattr__(self, name):
+        """Delete attributes on the wrapped modulestore"""
+        delattr(self._modulestore, name)
 
-        Arguments:
-          course (CourseModule or None)
+    def _wrap(self, block):
+        """Apply XBlock field data overrides."""
+        block._field_data = OverrideModulestoreFieldData.wrap(block, block._field_data)  # pylint: disable=protected-access
+        return block
 
-        Returns:
-          bool
-        """
-        return False
+    def get_item(self, usage_key, depth=0, **kwargs):
+        return self._wrap(
+            self._modulestore.get_item(usage_key, depth, **kwargs)
+        )
 
+    def get_items(self, course_key, **kwargs):
+        items = [self._wrap(item) for item in self._modulestore.get_items(course_key, **kwargs)]
+        return items
 
-def _lineage(block):
-    """
-    Returns an iterator over all ancestors of the given block, starting with
-    its immediate parent and ending at the root of the block tree.
-    """
-    parent = block.get_parent()
-    while parent:
-        yield parent
-        parent = parent.get_parent()
+    def get_course(self, course_key, depth=0, **kwargs):
+        return self._wrap(
+            self._modulestore.get_course(course_key, depth=depth, **kwargs)
+        )
+
+    def get_courses(self, **kwargs):
+        courses = [self._wrap(course) for course in self._modulestore.get_courses(**kwargs)]
```

Comment:
@cpennington that's correct. @jimabramson this is just mapping across an iterable returned by `get_courses`, so it doesn't change the Mongo call count. If it did, I'd expected to see tests failing all over the place.